wger/exercises/models.py

```diff
@@ -77,6 +77,26 @@ def get_owner_object(self):
         '''
         return False
 
+    def delete(self, *args, **kwargs):
+        """
+        Reset all cached info on muscle delete
+        """
+        for language in Language.objects.all():
+            delete_template_fragment_cache("exercise-overview", language.id)
+            delete_template_fragment_cache(
+                "exercise-overview-mobile", language.id
+            )
+            delete_template_fragment_cache(
+                "muscle-overview", language.id
+            )
+            delete_template_fragment_cache(
+                "equipment-overview", language.id
+            )
+        muscle_exercises = Exercise.objects.filter(muscles=self).iterator()
+        for exercise in muscle_exercises:
+            cache.clear()
```

Comment:
@Rhytah at this point can't we now use something like `cache.delete(cache_mapper.get_exercise_muscle_bg_key(exercise.id))` and it works? Because, since `cache.clear()` clears whole cache at once on the first iteration, the rest are redundant when it comes to clearing cache.

Reply:
@AGMETEOR I used it and it does not work to fulfil the desired outcome of the story as the muscle keeps on appearing in the exercises. But let me revert to  `cache.delete(cache_mapper.get_exercise_muscle_bg_key(exercise.id))`

Reply:
@Rhytah if you are using cache.clear(), there is no need to iterate through the muscles since it clears all the cache regardless.

Reply:
@Manorlds-Eaglespark I was trying out implementing @AGMETEOR's suggestion and this has been resolved.